homeassistant/components/abode/__init__.py

```diff
@@ -131,17 +129,50 @@ def is_light(self, device):
         )
 
 
-def setup(hass, config):
+async def async_setup(hass, config):
     """Set up Abode component."""
-    from abodepy.exceptions import AbodeException
+    if DOMAIN not in config:
+        return True
+
+    hass.data[DOMAIN] = {}
 
     conf = config[DOMAIN]
-    username = conf.get(CONF_USERNAME)
-    password = conf.get(CONF_PASSWORD)
-    name = conf.get(CONF_NAME)
-    polling = conf.get(CONF_POLLING)
-    exclude = conf.get(CONF_EXCLUDE)
-    lights = conf.get(CONF_LIGHTS)
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_USERNAME: conf.get(CONF_USERNAME),
+                CONF_PASSWORD: conf.get(CONF_PASSWORD),
+                CONF_NAME: conf.get(CONF_NAME),
+                CONF_POLLING: conf.get(CONF_POLLING),
+                CONF_EXCLUDE: conf.get(CONF_EXCLUDE),
+                CONF_LIGHTS: conf.get(CONF_LIGHTS),
+            },
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up Abode component via config entry (Integrations UI)."""
+    from abodepy.exceptions import AbodeException
+
+    username = config_entry.data.get(CONF_USERNAME)
+    password = config_entry.data.get(CONF_PASSWORD)
+    name = config_entry.data.get(CONF_NAME)
+    polling = config_entry.data.get(CONF_POLLING)
+    exclude = config_entry.data.get(CONF_EXCLUDE)
+    lights = config_entry.data.get(CONF_LIGHTS)
+
+    # 'exclude' and 'lights' has to be iterable, can't be of type None
+    if not exclude:
+        exclude = []
+
+    if not lights:
+        lights = []
 
     try:
         cache = hass.config.path(DEFAULT_CACHEDB)
```

Comment:
I'm struggling with this one. I thought I could simply utilize `async_add_executor_job` when the `AbodeSystem` instance is being created here:

`hass.data[DOMAIN] = AbodeSystem(username, password, cache, name, polling)`

This would turn into:

```
hass.data[DOMAIN] = hass.async_add_executor_job(
            AbodeSystem, username, password, cache, name, polling
        )
```
However, this doesn't appear to be working as the `AbodeSystem` instance isn't  being created.

I'm sure I'm messing up something fairly simple here.

Reply:
Deleted my previous message. I think I understand this now. I basically need to call `hass.async_add_executor_job` when defining `self.abode` in the `AbodeSystem` class.

Reply:
When I try using:

```
hass.data[DOMAIN] = hass.async_add_executor_job(
            AbodeSystem, username, password, cache, name, polling
        )
```

It appears `hass.data[DOMAIN]` is not being setup properly causing errors in every other method or module that references it.

Reply:
We need to await the task to get the result.
```py
await hass.async_add_executor_job(...)
```

Reply:
So that was the first thing I tried yesterday and I could have sworn it didn't work... of course when I try it again now after you mention it, it seems to be working fine. That was about two hours of my life... lol. Thanks again